app/schemas/models/user.coffee

```diff
@@ -417,6 +418,36 @@ _.extend UserSchema.properties,
         additionalProperties: FeatureRecipientSchema
 
   archived: c.date {description: 'Marks this record for automatic online archiving to cold storage by our cloud database.'}
+  products: c.array {title: 'Products purchased or used by this user'},
+    c.object { required: ['product', 'startDate', 'recipient', 'paymentService', 'paymentDetails'],  additionalProperties: true },
+      # ensure we can add additionalProperties
+      product: { type: 'string', enum: ['course', 'basic_subscription', 'pd', 'ai-league', 'online-classes'], decription: 'The "name" field for the product purchased' }  # And/or the ID of the Product in the database, if we make a Product for each thing we can buy?
+
+      prepaid: c.objectId(links: [ {rel: 'db', href: '/db/prepaid/{($)}'} ])  # required for type: “course” for legacy compatibility, optional for other types
+      productOptions:
+        oneOf: [
+          includedCourseIDs: {type: ['array', 'null']}
+        ]
+      startDate: c.date()
+      endDate: c.date()  # TODO: optional indication of no end date (lasts forever) - or do we just leave unset?
```

Comment:
unset is better

Reply:
yeah, while i don't really like the forever products. do we still have such products?

Reply:
I cant think of any from the ones i have dealt with 😅